runtime/hydradx/src/governance/tracks.rs

```diff
@@ -0,0 +1,206 @@
+//                $$$$$$$      Licensed under the Apache License, Version 2.0 (the "License")
+//             $$$$$$$$$$$$$        you may only use this file in compliance with the License
+//          $$$$$$$$$$$$$$$$$$$
+//                      $$$$$$$$$       Copyright (C) 2021-2024  Intergalactic, Limited (GIB)
+//         $$$$$$$$$$$   $$$$$$$$$$                       SPDX-License-Identifier: Apache-2.0
+//      $$$$$$$$$$$$$$$$$$$$$$$$$$
+//   $$$$$$$$$$$$$$$$$$$$$$$        $                      Built with <3 for decentralisation
+//  $$$$$$$$$$$$$$$$$$$        $$$$$$$
+//  $$$$$$$         $$$$$$$$$$$$$$$$$$      Unless required by applicable law or agreed to in
+//   $       $$$$$$$$$$$$$$$$$$$$$$$       writing, software distributed under the License is
+//      $$$$$$$$$$$$$$$$$$$$$$$$$$        distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+//      $$$$$$$$$   $$$$$$$$$$$         OR CONDITIONS OF ANY KIND, either express or implied.
+//        $$$$$$$$
+//          $$$$$$$$$$$$$$$$$$            See the License for the specific language governing
+//             $$$$$$$$$$$$$                   permissions and limitations under the License.
+//                $$$$$$$
+//                                                                 $$
+//  $$$$$   $$$$$                    $$                       $
+//   $$$     $$$  $$$     $$   $$$$$ $$  $$$ $$$$  $$$$$$$  $$$$  $$$    $$$$$$   $$ $$$$$$
+//   $$$     $$$   $$$   $$  $$$    $$$   $$$  $  $$     $$  $$    $$  $$     $$   $$$   $$$
+//   $$$$$$$$$$$    $$  $$   $$$     $$   $$        $$$$$$$  $$    $$  $$     $$$  $$     $$
+//   $$$     $$$     $$$$    $$$     $$   $$     $$$     $$  $$    $$   $$     $$  $$     $$
+//  $$$$$   $$$$$     $$      $$$$$$$$ $ $$$      $$$$$$$$   $$$  $$$$   $$$$$$$  $$$$   $$$$
+//                  $$$
+
+//! Track configurations for governance.
+
+use super::*;
+use primitives::constants::{
+	currency::UNITS,
+	time::{HOURS, MINUTES},
+};
+const fn percent(x: i32) -> sp_arithmetic::FixedI64 {
+	sp_arithmetic::FixedI64::from_rational(x as u128, 100)
+}
+
+use pallet_referenda::Curve;
+const APP_LINEAR: Curve = Curve::make_linear(7, 7, percent(50), percent(100));
+const APP_LINEAR_FLAT: Curve = Curve::make_linear(4, 7, percent(50), percent(100));
+const APP_RECIP: Curve = Curve::make_reciprocal(1, 7, percent(80), percent(50), percent(100));
+const SUP_LINEAR: Curve = Curve::make_linear(7, 7, percent(0), percent(50));
+const SUP_RECIP: Curve = Curve::make_reciprocal(5, 7, percent(1), percent(0), percent(50));
+const SUP_FAST_RECIP: Curve = Curve::make_reciprocal(3, 7, percent(1), percent(0), percent(50));
+const SUP_WHITELISTED_CALLER: Curve = Curve::make_reciprocal(1, 28, percent(20), percent(5), percent(50));
+
+const TRACKS_DATA: [(u16, pallet_referenda::TrackInfo<Balance, BlockNumber>); 9] = [
+	(
+		0,
+		pallet_referenda::TrackInfo {
+			name: "root",
+			max_deciding: 1,
+			decision_deposit: 5_000_000 * UNITS,
+			prepare_period: 24 * HOURS,
```

Comment:
we need more reasonable periods if we want to use this for runtime upgrades, currently only prepare + confirm + enactment will take 3 days

not even sure why is any prepare period necessary